x-pack/plugin/sql/connectors/tableau/connector/connectionProperties.js

```diff
@@ -0,0 +1,23 @@
+(function propertiesbuilder(attr) {
+    var props = {};
+
+    props["user"] = attr["username"];
+    props["password"] = attr["password"];
+
+    var extraProps = attr[connectionHelper.attributeWarehouse];
+    if (extraProps != null && extraProps.trim().length > 0) {
+        var avps = extraProps.trim().split(/[\s&]/);
+        for (var i = 0; i < avps.length; i++) {
+            var tokens = avps[i].split("=");
```

Comment:
Question, to be more lenient shouldn't we trim the split tokens, to treat something like:
` param1  =   value1  `

Reply:
Space is accepted as a delimiter between the AVPs, as a friendlier alternative to `&`; i.e. both forms `a=b&c=d` and `a=b c=d` are permitted. Spaces would normally need to be encoded in a URI.
That's why the initial AVP splitting is done by `.split(/[\s&]/)`.
I've added a comment.

Reply:
Thx for explaining!